src/numeric-types.c

```diff
@@ -93,6 +83,102 @@ numeric_##type##_to_##gtype (numeric_##type num)                     \
     return data.v_##gtype;                                           \
 }
 
+typedef struct _NumericTypeInfo
+{
+    GType             type;
+    const gchar      *name;
+    gsize             width;
+    NumericByteOrder  byte_order;
```

Comment:
@rainwoodman If you can think of any other field that could be of use in Vast for example.

If we know the width and byte ordering, we could very likely generate the code for byte swapping.